aten/src/ATen/WrapDimUtils.h

```diff
@@ -72,4 +72,13 @@ static inline int64_t legacy_cat_wrap_dim(int64_t dim, TensorList tensors) {
   return dim;
 }
 
+// wrap negative dims in to_transform_dims
+static inline void wrap_dims(std::vector<int64_t>& to_transform_dims, int64_t tensor_total_dims) {
+  for (size_t i = 0; i < to_transform_dims.size(); i++) {
+    if (to_transform_dims[i] < 0) {
```

Comment:
hmm.. I agree it is not intuitive. Would you like to raise a case that breaks this function? I can't think of a good example, for instance:

```
(-4 % 3 + 3 ) % 3 = 2
```
This works because -4 equals to -1 in total_dim = 3, where index = 2 is equivalent to index = -1

Reply:
wait, I still feel like this works even with dim[i] < -total_dims. So in textbook, modulo in C defines as `a/b*b + a%b == a` -> `a%b == a - a/b*b`, and in the original formula, I took `a%b + b == a - a/b*b + b`. When `a > 0`, '(a%b + b) % b == a % b'. When `a < 0`, `a - a/b*b == remainder of a/b`, and I think (don't know how to prove) `b + remainder of a/b == a%b`. I guess we can verify that `a%b + b (in C) == a%b (in python)` when `a < -b`. Besides it passes test:

```
self.assertEqual(data.rot90(-5, [0, 1]), data.rot90(-1, [0, 1]))
```

Reply:
Yes it works. The point is that it shouldn't work when it is below `-tensor_total_dims`.

Reply:
See `maybe_wrap_dim` on top of this file.

Reply:
Let's just use that inline fn in this for loop.

Reply:
lazily checking dim out-of-bound for now since that inline fn needs a 2D vector for args...maybe I can use your implementation after code lands

Reply:
this needs a clearer name, because it's not obvious from the name that this wraps all the dimensions; it's difficult to distinguish between "dim" and "dims".  Call this wrap_all_dimensions or somethign.  Also I believe @SsnL was talking about the first function in the file, the 2D vector function isn't relevant here.

Reply:
ah, I see! Thanks! Will change accordingly